clients/typescript/src/satellite/process.ts

```diff
@@ -628,8 +685,36 @@ export class SatelliteProcess implements Satellite {
     }
   }
 
-  async _connectAndStartReplication(): Promise<void> {
-    Log.info(`connecting and starting replication`)
+  private initializing?: {
+    promise: Promise<void>
+    resolve: () => void
+    reject: (e?: unknown) => void
+  }
+
+  async _connectWithBackoff() {
+    this.initializing = emptyPromise()
+
+    await backOff(() => this._connect(), {
+      ...connectionRetryPolicy,
+      retry: connectRetryHandler,
+    }).catch((e) => {
+      // We're very sure that no calls are going to modify `this.initializing` before this promise resolves
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+
+      const error = new SatelliteError(
+        SatelliteErrorCode.CONNECTION_FAILED_AFTER_RETRY,
+        `Failed to connect to server after exhausting retry policy. Last error thrown by server: ${e.message}`
+      )
+
+      this.initializing?.reject(error)
+      this.notifier.connectivityStateChanged(this.dbName, 'disconnected')
+      throw error
+    })
+  }
+
+  // NO DIRECT CALLS TO CONNECT
+  async _connect(): Promise<void> {
```

Comment:
Then make it private :D


```suggestion
  private async _connect(): Promise<void> {
```

Reply:
sure :). I had to remove a bunch of internal calls to connect too.